waffle_hub/hub/hub.py

```diff
@@ -402,6 +407,51 @@ def get_hub_list(cls, root_dir: str = None) -> list[str]:
                     hub_name_list.append(hub_dir.name)
         return hub_name_list
 
+    @classmethod
+    def from_waffle_file(cls, name: str, waffle_file: str, root_dir: str = None) -> "Hub":
+        """Import new Hub with waffle file for inference.
+
+        Args:
+            name (str): hub name.
+            waffle_file (str): waffle file path.
+            root_dir (str, optional): hub root directory. Defaults to None.
+
+        Returns:
+            Hub: New Hub instance
+        """
+        root_dir = Hub.parse_root_dir(root_dir)
+
+        if name in cls.get_hub_list(root_dir):
+            raise FileExistsError(f"{name} already exists. Try another name.")
+
+        if not os.path.exists(waffle_file):
+            raise FileNotFoundError(f"Waffle file {waffle_file} is not exist.")
+
+        if os.path.splitext(waffle_file)[1] != ".waffle":
+            raise ValueError(
+                f"Invalid waffle file: {waffle_file}, Waffle File extension must be .waffle."
+            )
+
```

Comment:
It's my mistake..
I will modify it later because it is against the code conventions.